File: src/test/programmers/python/level3/Network.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    for i in range(n):
        if visited[i] is False:
            dfs(n, computers, visited, i)
            answer += 1
    return answer

def dfs(n, computers, visited, i):
    visited[i] = True
    for j in range(n):
        if visited[j] is False and computers[i][j] == 1:
            dfs(n, computers, visited, j)

File: src/test/programmers/java/level3/Network.java
package test.programmers.java.level3;

public class Network {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];

        for(int i = 0; i < n; i++) {
            if(visited[i] == false) {
                dfs(n, computers, visited, i);
                answer++;
            }
        }

        return answer;
    }

    static void dfs(int n, int[][] computers, boolean[] visited, int i) {
        visited[i] = true;

        for(int j = 0; j < n; j++) {
            if(visited[j] == false && computers[i][j] == 1) {
                dfs(n, computers, visited, j);
            }
        }
    }
}
